cmd/build/v2/smartbuild/hasher_test.go

```diff
@@ -20,8 +20,38 @@ import (
 	"github.com/okteto/okteto/pkg/build"
 	"github.com/spf13/afero"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 )
 
+type fakeWorkingDirGetter struct {
+	err        error
+	workingDir string
+}
+
+func (f fakeWorkingDirGetter) Get() (string, error) {
+	if f.err != nil {
+		return "", f.err
+	}
+	return f.workingDir, nil
+}
+
+type mockConfigRepo struct {
+	mock.Mock
+}
```

Comment:
Added this mock to be able to do assertions based on the build context passed to the repoCtrl, and verify we are passing the expected values

Reply:
😮 i did not knew thisss